src/workbookManager.ts

```diff
@@ -34,31 +36,97 @@ export class WorkbookManager {
                   )
                 : await JSZip.loadAsync(templateFile);
 
-        return await this.generateSingleQueryWorkbookFromZip(zip, query);
+        return await this.generateQuery1WorkbookFromZip(zip, query, docProps);
     }
 
-    private async generateSingleQueryWorkbookFromZip(
+    private async generateQuery1WorkbookFromZip(
         zip: JSZip,
-        query: QueryInfo
+        query: QueryInfo,
+        docProps?: DocProps
     ): Promise<Blob> {
+        await this.updatePowerQueryDocument(zip, query.queryMashup);
+        await this.updateSingleQueryRefreshOnOpen(zip, query.refreshOnOpen);
+        await this.updateDocProps(zip, docProps);
+
+        return await zip.generateAsync({
+            type: "blob",
+            mimeType: "application/xlsx",
+        });
+    }
+
+    private async updatePowerQueryDocument(zip: JSZip, queryMashup: string) {
+        const mashupHandler = new MashupHandler();
         const old_base64 = await pqUtils.getBase64(zip);
-        const new_base64 = await this.mashupHandler.ReplaceSingleQuery(
+        const new_base64 = await mashupHandler.ReplaceSingleQuery(
             old_base64,
-            query.queryMashup
+            queryMashup
         );
         await pqUtils.setBase64(zip, new_base64);
+    }
 
-        if (query.refreshOnOpen) {
-            await this.setSingleQueryRefreshOnOpen(zip);
-        }
+    private async updateDocProps(zip: JSZip, docProps: DocProps = {}) {
 
-        return await zip.generateAsync({
-            type: "blob",
-            mimeType: "application/xlsx",
+        //set defaults
+        if (!docProps.title) docProps.title = defaultDocProps.title;
+        if (!docProps.createdBy) docProps.createdBy = defaultDocProps.createdBy;
+        if (!docProps.lastModifiedBy)
+            docProps.lastModifiedBy = defaultDocProps.lastModifiedBy;
+
+            //set auto updated elements
+        const { doc, properties } = await documentUtils.getDocPropsProperties(
+            zip
+        );
+
+        const docPropsAutoUpdatedElementsArr = Object.keys(
```

Comment:
It would be executed only once, since this api just generates a new workbook.
 I updated the dates in the new default template PR, but I think it would still be better if every time we generate a workbook we use current time of generation.